### PR TITLE
HomeWork 2

### PR DESCRIPTION
**Pull request by andrei-mikhailovskii**
add AbstractChromeTest, ExerciseOne, ExerciseTwo classes 
add Maven profile called 'hw2'

### CODE REVIEW

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    @Test
+    public void login() {
+
+        try {
+            new WebDriverWait(driver, 5)
+                    .until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("li.uui-profile-menu"))).click();
+            driver.findElement(By.id("name")).sendKeys("Roman");
+            driver.findElement(By.id("password")).sendKeys("Jdi1234");
+            driver.findElement(By.id("login-button")).click();
+        } catch (NoSuchElementException e) {
```
Comment (knadonenkoepam) [inline src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: That's actually not a good practice to try and catch inside the test

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+import java.util.List;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.Test;
+
+public class ExerciseOneTest extends AbstractChromeTest {
```
Comment (knadonenkoepam) [inline src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java:12]: Every step should be under one test method

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    //Assert that there are 5 items in the Left Section are displayed, and they have proper text
+    @Test
+    public void leftSection() {
+
+        SoftAssertions softly = new SoftAssertions();
+        List<WebElement> searchResults = driver.findElements(By.cssSelector(".sidebar-menu>li"));
+        softly.assertThat(searchResults.size() == 5).as("There are not 5 elements in the Left Side section").isTrue();
+
+        softly.assertThat(driver.findElement(By.cssSelector("ul.sidebar-menu.left > li[index='1']"))
```
Comment (knadonenkoepam) [inline src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: Locators are in a bad shape, they could be simplified and be more readable

File: src/test/java/com/epam/tc/hw2/ex2/ExerciseTwoTest.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            driver.findElement(By.id("name")).sendKeys("Roman");
+            driver.findElement(By.id("password")).sendKeys("Jdi1234");
+            driver.findElement(By.id("login-button")).click();
+        } catch (NoSuchElementException e) {
+            System.err.println("No element: " + e);
+        }
+        softly.assertThat(driver.findElement(By.id("user-name")).getText()).isEqualTo("ROMAN IOVLEV");
+
+        //Open through the header menu Service -> Different Elements Page
+        driver.findElement(By.cssSelector("ul.m-l8>li>a.dropdown-toggle")).click();
```
Comment (knadonenkoepam) [inline src/test/java/com/epam/tc/hw2/ex2/ExerciseTwoTest.java]: could be simplified :) hint: it's a link

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+    //Assert that there are 4 texts on the Index Page under icons and they have proper text
+    @Test
+    public void textCheck() {
+
+        SoftAssertions softly = new SoftAssertions();
+        List<WebElement> searchResults = driver.findElements(By.className("benefit-txt"));
+        softly.assertThat(searchResults.size() == 4).as("Texts quantity inconsistency").isTrue();
+
+        softly.assertThat(driver
```
Comment (knadonenkoepam) [inline src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: You don't need the whole text here :)

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        softly.assertAll();
+
+    }
+
+    //Assert that there are 4 items on the header section are displayed and they have proper texts
+    @Test
+    public void navBarCheck() {
+
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(driver.findElement(By.cssSelector("ul.m-l8>li>a[href='index.html']")).isDisplayed());
```
Comment (knadonenkoepam) [inline src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: Could be simblified and maybe you don't need a css selector right here

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java (hunk 1 above)
Comment (andrei-mikhailovskii) [reply src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: try-catch construction is removed

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java (hunk 2 above)
Comment (andrei-mikhailovskii) [reply src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java:12]: ExerciseOneTest.java is refactored, and now consists of one method where all steps are gathered

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java (hunk 3 above)
Comment (andrei-mikhailovskii) [reply src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: the whole check is modified, and now performed via loop, which is more simple construction, and unnecessary locators are removed

File: src/test/java/com/epam/tc/hw2/ex2/ExerciseTwoTest.java (hunk 4 above)
Comment (andrei-mikhailovskii) [reply src/test/java/com/epam/tc/hw2/ex2/ExerciseTwoTest.java]: the locator is modified and simplified

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java (hunk 5 above)
Comment (andrei-mikhailovskii) [reply src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: check is modified, unnecessary selectors are removed, check is now performed via "for" loop

File: src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java (hunk 6 above)
Comment (andrei-mikhailovskii) [reply src/test/java/com/epam/tc/hw2/ex1/ExerciseOneTest.java]: simplified, some css locators are replaced with By.linkText() ones

**Review by knadonenkoepam — APPROVED** [review]

**Pull request merged**